hs_core/hydroshare/utils.py

```diff
@@ -682,6 +684,60 @@ def validate_resource_file_count(resource_cls, files, resource=None):
                 raise ResourceFileValidationException(err_msg)
 
 
+def convert_file_size_to_unit(size, unit):
+    """
+    Convert file size to unit for quota comparison
+    :param size: in byte unit
+    :param unit: should be one of the four: 'KB', 'MB', 'GB', or 'TB'
+    :return: the size converted to the pass-in unit
+    """
+    unit = unit.lower()
+    kbsize = size / 1000.0
+    if unit == 'kb':
+        return kbsize
+    mbsize = kbsize / 1000.0
+    if unit == 'mb':
+        return mbsize
+    gbsize = mbsize / 1000.0
+    if unit == 'gb':
+        return gbsize
+    tbsize = gbsize / 1000.0
+    if unit == 'tb':
+        return tbsize
+
+    raise ValidationError('Pass-in unit for file size conversion must be one of KB, MB, GB, or TB')
```

Comment:
@pkdash OK, will move raising validation exception to the beginning, which would make the logic a bit clearer.